src/lib/apparmor/genprof.rb

```diff
@@ -0,0 +1,45 @@
+
+require 'yast'
+require 'apparmor/apparmor_ui_dialog'
+
+Yast.import 'UI'
+Yast.import 'Label'
+Yast.import 'Popup'
+
+module AppArmor
+  # GenProf class executes the aa-genprof command and generates
+  # the appropriate dialogs for the yast UI
+  class GenProf < AAProgram
+    include Yast::UIShortcuts
+    include Yast::Logger
+    include Yast::I18n
+
+    def initialize
+      textdomain 'apparmor'
+      @program = Yast::UI.AskForExistingFile('/usr/bin/', '*',
+                                             _('Choose a program to generate a profile for'))
+      return nil if @program.nil?
```

Comment:
this won't work. when you return from initialize do nothing, because method new always return new instance of class.

try it yourself in irb ( ruby interactive shell):

```
irb(main):001:0> class C; def initialize; return nil; end; end
=> :initialize
irb(main):002:0> C.new
=> #<C:0x000000018dd258>
```